base4kids2.ldif

```diff
@@ -498,6 +498,43 @@ attributeTypes: (
       DESC 'Determines whether a base4kids object is disabled but still active'
       SUP base4kidsIsBlocked )
 #
+# Name:     base4kidsTelephoneNumberBusiness
+# Syntax:   Phone number
+# Examples: +01234567890
+attributeTypes: (
+      1.3.6.1.4.1.31534.2.5.2.42
+      NAME 'base4kidsTelephoneNumberBusiness'
+      DESC 'Business telephone number'
+      EQUALITY telephoneNumberMatch
+      ORDERING telephoneNumberOrderingMatch
+      SYNTAX 1.3.6.1.4.1.1466.115.121.1.50
+      SINGLE-VALUE )
+#
+# Name:     base4kidsDateOfBirth
+# Syntax:   Generalized Time
+# Examples: 19951221000000Z
+attributeTypes: (
+      1.3.6.1.4.1.31534.2.5.2.43
+      NAME 'base4kidsDateOfBirth'
+      DESC 'Date of Birth'
+      EQUALITY generalizedTimeMatch
+      ORDERING generalizedTimeOrderingMatch
+      SYNTAX 1.3.6.1.4.1.1466.115.121.1.24
+      SINGLE-VALUE )
+#
+# Name:     base4kidsPlaceOfOrigin
+# Syntax:   Directory String
+# Examples: "Bern"
+attributeTypes: (
+      1.3.6.1.4.1.31534.2.5.2.44
+      NAME 'base4kidsPlaceOfOrigin'
+      DESC 'The name of a place (city)'
+      EQUALITY caseExactMatch
```

Comment:
Please don't redefine the syntax and matchings rules but rather simply derive it from the existing `l` (localityName) attribute (RFC 2256 ): `SUP l`

Reply:
OK, I'll do that. However, in my opinion, it's a very good idea when we have a group of attributes which should always have the same template, so in order to change type we just do it in one place. Putting some accidental similar template may occur the delayed ignition bomb, in some more complex schema (the template definition is changed for some feature and exceptions appear on some other attributes).

Reply:
I agree that one has to be careful with attribute derivations. However, in this case the place of origin matches 1:1 with the locality name (_this attribute contains the name of a locality, such as a city [...]_) which is very unlikely to change as it has been defined ages ago within RFC 2256 and various implementation depend on it.

Reply:
Yes, place of origin is not a good example - it's just a city-like string so "l" seems to be a good template. I wrote generally about type derrivation - the bad example is date of birth mentioned in another comment.